apps/web/vibes/soul/form/radio-group/index.tsx

```diff
@@ -18,37 +18,13 @@ export const RadioGroup = React.forwardRef<
     options: Option[];
     errors?: string[];
   }
->(({ id, label, options, errors, className, ...rest }, ref) => {
+>(({ label, options, errors, className, ...rest }, ref) => {
   return (
     <div className={clsx('space-y-2', className)}>
-      {label !== undefined && label !== '' && <Label htmlFor={id}>{label}</Label>}
-      <RadioGroupPrimitive.Root
-        {...rest}
-        aria-label={label}
-        className="space-y-2"
-        id={id}
-        ref={ref}
-      >
-        {options.map((option) => (
-          <div className="flex items-center" key={option.value}>
-            <RadioGroupPrimitive.Item
-              aria-label={option.label}
-              className={clsx(
-                'size-5 cursor-default rounded-full border bg-background outline-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50 [&:disabled+label]:pointer-events-none [&:disabled+label]:opacity-50',
-                errors && errors.length > 0
-                  ? 'border-error disabled:border-error/50'
-                  : 'border-contrast-200 hover:border-contrast-300 focus:border-contrast-300',
-              )}
-              disabled={option.disabled}
-              id={option.value}
-              value={option.value}
-            >
-              <RadioGroupPrimitive.Indicator className="relative flex size-full items-center justify-center after:block after:size-3 after:rounded-full after:bg-foreground" />
-            </RadioGroupPrimitive.Item>
-            <label className="pl-3 text-sm leading-none text-foreground" htmlFor={option.value}>
-              {option.label}
-            </label>
-          </div>
+      {label !== undefined && label !== '' && <Label>{label}</Label>}
+      <RadioGroupPrimitive.Root {...rest} aria-label={label} className="space-y-2" ref={ref}>
+        {options.map((option, index) => (
+          <RadioGroupItem error={errors != null && errors.length > 0} key={index} option={option} />
```

Comment:
What happened to using `value` as the `key`?

Reply:
There's no enforcement of value uniqueness in the API and I'd rather it not blow up and just show the duplicate option. AFAIK the main reason to not use index would be performance reasons right?